src/NServiceBus.Transport.RabbitMQ/RabbitMQTransportInfrastructure.cs

```diff
@@ -122,13 +122,23 @@ public override Task Stop()
 
         IRoutingTopology CreateRoutingTopology()
         {
-            var durable = settings.DurableMessagesEnabled();
-
             if (!settings.TryGet(out Func<bool, IRoutingTopology> topologyFactory))
             {
                 throw new InvalidOperationException("A routing topology factory has not been configured. Configure a routing topology with a TransportExtensions<RabbitMQTransport>.UseXXXRoutingTopology() method.");
             }
 
+            if (!settings.TryGet(SettingsKeys.UseDurableExchangesAndQueues, out bool durable))
```

Comment:
> > We have the new setting, and the throwing of the exception, but don't we actually need to do something to use that new setting to control the durability of the exchanges and queues separately from the durability of the messages?
> 
> @adamralph I'm not sure I understand this comment. The change I've made [here](https://github.com/Particular/NServiceBus.RabbitMQ/pull/443/files#diff-179646131f3010c16f5abb4c584dfb24R130) means that we're using the value from the new setting to control durability.

Apologies, I misread things. I assumed that the `durable` variable referred to the entirety of "durability" i.e. messages, exchanges and queues, and forgot that core is taking care of message durability, and the transport only controls exchange and queue durability.